src/libsync/propagateupload.cpp

```diff
@@ -676,6 +676,44 @@ void PropagateUploadFileCommon::commonErrorHandling(AbstractNetworkJob *job)
     SyncFileItem::Status status = classifyError(job->reply()->error(), _item->_httpErrorCode,
         &propagator()->_anotherSyncNeeded, replyContent);
 
+    // Remote host closed the connection after an upload larger than _failsafeMaxChunkSize
+    if (job->reply()->error() == QNetworkReply::RemoteHostClosedError 
+        && _item->_requestBodySize > propagator()->syncOptions()._failsafeMaxChunkSize) {
+        // treat it as if the server returned HTTP-413
+        qCInfo(lcPropagateUpload()) << "Remote host closed the connection when uploading" 
+                                    << Utility::octetsToString(_item->_requestBodySize) << "."
+                                    << "Treating the error like HTTP-413 (request entity to large)";
+
+        _item->_httpErrorCode = 413;
```

Comment:
Best is not to override `_item->_httpErrorCode`, but add this `job->reply()->error() == QNetworkReply::RemoteHostClosedError 
        && _item->_requestBodySize > propagator()->syncOptions()._failsafeMaxChunkSize` as a part of an `OR` condition in the `if (_item->_httpErrorCode == 413 && _item->_requestBodySize > 0 && status == SyncFileItem::NormalError)`.

Reply:
Refactored this part so that the response code is now longer changed.